src/components/tree.js

```diff
@@ -32,155 +37,138 @@ export default class Tree extends Component {
       source: "lol.not",
       pros: [],
       cons: []
-    }
+    };
 
     this.state = {
-      // the argument that we're representing
-      tree: defaultArgument,
-      // list of types (pro/con, index)
-      path: [],
       // loading icon indication
       loading: false
-    }
+    };
 
-    this.saveTree = this.saveTree.bind(this)
-    this.advancePath = this.advancePath.bind(this)
-    this.regressPath = this.regressPath.bind(this)
-    this.setConfidence = this.setConfidence.bind(this)
-    this.setDescription = this.setDescription.bind(this)
-    this.setSource = this.setSource.bind(this)
-    this.setTitle = this.setTitle.bind(this)
-    this.addStatement = this.addStatement.bind(this)
-    this.removeStatement = this.removeStatement.bind(this)
+    this.setCurrentStatement = this.setCurrentStatement.bind(this);
+    // this.changeCurrentStatement = thiss.changeCurrentStatement.bind(this);
+
+    this.saveTree = this.saveTree.bind(this);
+    this.setConfidence = this.setConfidence.bind(this);
+    this.setDescription = this.setDescription.bind(this);
+    this.setSource = this.setSource.bind(this);
+    this.setTitle = this.setTitle.bind(this);
+    this.addStatement = this.addStatement.bind(this);
+    this.removeStatement = this.removeStatement.bind(this);
+    this.handleData = this.handleData.bind(this);
+
+    
+  }
+
+  handleData(data){
+    this.props.actions.updateTree(data);
   }
 
   componentDidMount() {
     fetch("/api")
       .then(res => res.json())
-      .then(data => this.setState({tree: data}))
-      .catch(() => console.log("Could not fetch data =("))
+      .then(data => this.handleData(data))
+      .catch(() => console.log("Could not fetch data =("));
   }
 
   saveTree() {
     fetch('/api/tree', {
       credentials: 'same-origin',
-      body: JSON.stringify(this.state.tree),
+      body: JSON.stringify(this.props.tree),
       method: 'POST',
       headers: {
         'Content-Type': 'application/json'
       }
-    }).then(res => console.log(res.status))
+    }).then(res => console.log(res.status));
   }
 
-  advancePath(pro, index) {
-    this.setState({
-      path: this.state.path.concat((pro ? "pros" : "cons") + index)
-    })
-  }
+  // refactor so you just get the statements to display
+  // get current statement
 
-  regressPath(amt) {
-    this.setState({
-      path: this.state.path.slice(0, amt)
-    })
-  }
+  // use a sort of map thingy
+  setCurrentStatement(property, val){
```

Comment:
This code has *finally* been refactored and centralized into one function. Nice work! 🎉 😁